NativeApp/src/components/Booking/container.js

```diff
@@ -126,14 +132,27 @@ export default Container => class extends Component {
       ));
   }
 
+  updateHalfDay = () => {
+    const { booking } = this.state;
+    const { halfDay, startDate } = booking;
+
+    this.setState({
+      booking: {
+        ...booking,
+        endDate: startDate,
+        halfDay: !halfDay,
+      },
+    });
+  }
+
   render() {
     const { booking, booked } = this.state;
```

Comment:
Destructure booking here too. Normally I don't mind this but you've done it in `updateHalfDay` just above.

Reply:
Don't think there would be much point destructuring booking. It's being passed to the View, instead of passing the values individually.